### PR TITLE
HAAR-498: hide download link

### PR DESCRIPTION
**Pull request by ravmoj**
hide download link based on number of records

### CODE REVIEW

File: backend/controllers/searchWithFilter.test.js [hunk 1]
```diff
@@ -38,10 +39,11 @@ describe('search factory', () => {
       getCaseloadsApi.mockReset()
       getSearchableRolesApi.mockResolvedValue([{ roleName: 'Access Role Admin', roleCode: 'ACCESS_ROLE_ADMIN' }])
       getCaseloadsApi.mockResolvedValue([{ text: 'Moorland HMP', value: 'MDI' }])
-      findUsersApi.mockResolvedValue({ searchResults: [], number: 0, page: 0 })
+      findUsersApi.mockResolvedValue({ searchResults: [], totalElements: 20, number: 0, page: 0 })
```
Comment (brightonsbox) [inline backend/controllers/searchWithFilter.test.js:42]: Maybe should be 0 so it matches the search results array size?

File: views/searchWithFilter.njk [hunk 2]
```diff
@@ -208,6 +208,8 @@
                   <div class="pull-right">
                     <a href="{{ downloadUrl }}" class="govuk-link" data-qa="download" download="user-search.csv" type="text/csv">Download results</a>
                   </div>
+                  {% else %}
```
Comment (brightonsbox) [inline views/searchWithFilter.njk:211]: Won't this message also show if the `allowDownload(res)` method returns false? I think you need an admin role to see the download link so if you didn't have the admin role you'd potentially see a misleading message. Might need to pass through another boolean into here such as `exceededDownloadRecordLimit` or similar

Comment (ravmoj) [reply views/searchWithFilter.njk]: Changed to hide download link and message to non-admin users

File: backend/controllers/searchWithFilter.test.js (hunk 1 above)
Comment (ravmoj) [reply backend/controllers/searchWithFilter.test.js]: changed to zero

File: backend/config.js [hunk 3]
```diff
@@ -57,4 +57,5 @@ module.exports = {
     dpsUserOldSearch: process.env.FEATURE_DPS_USER_OLD_SEARCH !== 'false',
   },
   phaseName: process.env.SYSTEM_PHASE,
+  downloadRecordLimit: toInt(process.env.BETA_SEARCH_DOWNLOAD_LINK_LIMIT, 20000),
```
[the remark this answers is not in the document]
Comment (ravmoj) [reply backend/config.js:60]: Changed to DPS_SEARCH_DOWNLOAD_LINK_LIMIT

**Review by brightonsbox — APPROVED** [review]
LGTM

**Pull request merged**